hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieSnapshotExporter.java

```diff
@@ -0,0 +1,233 @@
+    String snapshotPrefix;
+
+    @Parameter(names = {"--output-format"}, description = "e.g. Hudi or Parquet", required = true)
+    String outputFormat;
+
+    @Parameter(names = {"--output-partition-field"}, description = "A field to be used by Spark repartitioning")
+    String outputPartitionField;
+  }
+
+  public int export(SparkSession spark, Config cfg) throws IOException {
+    JavaSparkContext jsc = new JavaSparkContext(spark.sparkContext());
+    FileSystem fs = FSUtils.getFs(cfg.sourceBasePath, jsc.hadoopConfiguration());
+
+    final SerializableConfiguration serConf = new SerializableConfiguration(jsc.hadoopConfiguration());
+    final HoodieTableMetaClient tableMetadata = new HoodieTableMetaClient(fs.getConf(), cfg.sourceBasePath);
+    final TableFileSystemView.BaseFileOnlyView fsView = new HoodieTableFileSystemView(tableMetadata,
+        tableMetadata.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
+    // Get the latest commit
+    Option<HoodieInstant> latestCommit =
+        tableMetadata.getActiveTimeline().getCommitsTimeline().filterCompletedInstants().lastInstant();
+    if (!latestCommit.isPresent()) {
+      LOG.error("No commits present. Nothing to snapshot");
+      return -1;
+    }
+    final String latestCommitTimestamp = latestCommit.get().getTimestamp();
+    LOG.info(String.format("Starting to snapshot latest version files which are also no-late-than %s.",
+        latestCommitTimestamp));
+
+    List<String> partitions = FSUtils.getAllPartitionPaths(fs, cfg.sourceBasePath, false);
+    if (partitions.size() > 0) {
+      List<String> dataFiles = new ArrayList<>();
+
+      if (!StringUtils.isNullOrEmpty(cfg.snapshotPrefix)) {
+        for (String partition : partitions) {
+          if (partition.contains(cfg.snapshotPrefix)) {
+            dataFiles.addAll(fsView.getLatestBaseFilesBeforeOrOn(partition, latestCommitTimestamp).map(f -> f.getPath()).collect(Collectors.toList()));
+          }
+        }
+      } else {
+        for (String partition : partitions) {
+          dataFiles.addAll(fsView.getLatestBaseFilesBeforeOrOn(partition, latestCommitTimestamp).map(f -> f.getPath()).collect(Collectors.toList()));
+        }
+      }
+      try {
+        DataSource.lookupDataSource(cfg.outputFormat, spark.sessionState().conf());
+      } catch (Exception e) {
+        LOG.error(String.format("The %s output format is not supported! ", cfg.outputFormat));
+        return -1;
+      }
+      if (!cfg.outputFormat.equalsIgnoreCase("hudi")) {
+        // Do transformation
+        // A field to do simple Spark repartitioning
+        DataFrameWriter<Row> write = null;
+        Dataset<Row> original = spark.read().parquet(JavaConversions.asScalaIterator(dataFiles.iterator()).toSeq());
+        List<Column> needColumns = Arrays.asList(original.columns()).stream().filter(col -> !col.contains("_hoodie_")).map(col -> new Column(col)).collect(Collectors.toList());
```

Comment:
you need `startswith()` to be safe. There could be a column named "total_hoodie_counts"

Reply:
This is optional but actually i'd even prefer having an immutable list of `_hoodie_*` reserved field names so we can be explicit on what we are removing while exporting.